packages/perseus-core/src/widgets/matcher/matcher-util.test.ts

```diff
@@ -1,51 +1,177 @@
+            problemNum: 0,
+        };
+
+        const result = shuffleMatcher(options);
+        expect(result.left).toEqual(["1"]);
+        expect(result.right).toEqual(["A"]);
+    });
+
+    it("does nothing when all cards are the same", () => {
+        const options = {
+            orderMatters: true,
+            left: ["", ""],
+            right: ["", ""],
+            problemNum: 0,
+        };
+
+        const result = shuffleMatcher(options);
+        expect(result.left).toEqual(["", ""]);
+        expect(result.right).toEqual(["", ""]);
     });
+
+    it("shuffles the right column when orderMatters is false, using problemNum as the seed", () => {
+        const options = {
+            orderMatters: false,
+            left: ["1", "2", "3"],
+            right: ["A", "B", "C"],
+            problemNum: 0,
+        };
+
+        const result = shuffleMatcher(options);
+        expect(result).toEqual({left: ["1", "2", "3"], right: ["B", "A", "C"]});
+    });
+
+    describe.each(range(0, 30))(
+        "when orderMatters is false and problemNum is %d",
+        (problemNum) => {
+            const options = {
+                orderMatters: false,
+                left: ["1", "2", "3"],
+                right: ["A", "B", "C"],
+                problemNum,
+            };
+
+            it("displaces the first card in the right column", () => {
+                expect(shuffleMatcher(options).right[0]).not.toBe("A");
+            });
+
+            it("does not shuffle the left column", () => {
+                expect(shuffleMatcher(options).left).toEqual(["1", "2", "3"]);
+            });
+        },
+    );
+
+    it("shuffles both columns when orderMatters is true, using problemNum as the seed", () => {
+        const options = {
+            orderMatters: true,
+            left: ["1", "2", "3"],
+            right: ["A", "B", "C"],
+            problemNum: 0,
+        };
+
+        const result = shuffleMatcher(options);
+        expect(result).toEqual({left: ["2", "1", "3"], right: ["B", "A", "C"]});
+    });
+
+    describe.each(range(0, 30))(
+        "when orderMatters is true and problemNum is %d",
+        (problemNum) => {
+            const options = {
+                orderMatters: true,
+                left: ["1", "2", "3"],
+                right: ["A", "B", "C"],
+                problemNum,
+            };
+
+            it("displaces the first card in the left column", () => {
+                expect(shuffleMatcher(options).left[0]).not.toBe("1");
+            });
+
+            it("displaces the first card in the right column", () => {
+                expect(shuffleMatcher(options).right[0]).not.toBe("A");
+            });
+        },
+    );
 });
+
+function range(min: number, max: number): number[] {
+    invariant(min <= max, "range: min must be <= max");
+    return [
+        ...(function* () {
+            for (let n = min; n <= max; n++) {
+                yield n;
+            }
+        })(),
+    ];
```

Comment:
This is pretty clever, but I worry it would be confusing to some.

```suggestion
    const arr = []
        for (let i = min; i <= max; i++) {
            arr.push(i)
        }
    return arr
```

That's shorter and clearer IMO. There's also:

```suggestion
    return new Array(max-min+1)
        .fill()
        .map((_, i) => i + min)
```

But I like my first suggestion the most.